### PR TITLE
Allow AnkiDroid to register itself to open tsv/csv files

### LINKED ISSUE #15185: AnkiDroid doesn't register itself to open tsv/csv files
> Also, interestingly, AnkiDroid is not shown as an option when opening .csv and .tsv files, but it should be there in order to easily import these files.

Originally posted in https://github.com/ankidroid/Anki-Android/issues/15175#issue-2069704499

### PR DESCRIPTION
**Pull request by criticalAY**
## Purpose / Description
Allow AnkiDroid itself to register for tsv/csv files via intent

## Fixes
* Fixes #15185

## Approach
Create an intent with meme type csv/tsv

## How Has This Been Tested?

Tested on Google emulator  API33/24/23

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by criticalAY** [comment]
Not sure if we still need this or not as it worked for csv files without it also but leaving it for now and will act further as per review 

```
     <!-- tsv files -->
                <data
                    android:host="*"
                    android:mimeType="*/*"
                    android:pathPattern=".*\\.tsv"
                    android:scheme="http"
                    />
                <data
                    android:host="*"
                    android:mimeType="*/*"
                    android:pathPattern=".*\\.tsv"
                    android:scheme="https"
                    />
                <data
                    android:host="*"
                    android:mimeType="*/*"
                    android:pathPattern=".*\\.tsv"
                    android:scheme="content"
                    />
                <data
                    android:host="*"
                    android:mimeType="*/*"
                    android:pathPattern=".*\\.tsv"
                    android:scheme="file"
                    />
                <!-- csv files -->
                <data
                    android:host="*"
                    android:mimeType="*/*"
                    android:pathPattern=".*\\.csv"
                    android:scheme="http"
                    />
                <data
                    android:host="*"
                    android:mimeType="*/*"
                    android:pathPattern=".*\\.csv"
                    android:scheme="https"
                    />
                <data
                    android:host="*"
                    android:mimeType="*/*"
                    android:pathPattern=".*\\.csv"
                    android:scheme="content"
                    />
                <data
                    android:host="*"
                    android:mimeType="*/*"
                    android:pathPattern=".*\\.csv"
                    android:scheme="file"
                    />
```

File: AnkiDroid/src/main/java/com/ichi2/anki/IntentHandler.kt [hunk 1]
```diff
@@ -220,7 +231,11 @@ class IntentHandler : Activity() {
         fun getLaunchType(intent: Intent): LaunchType {
             val action = intent.action
             return if (Intent.ACTION_VIEW == action && isValidViewIntent(intent)) {
-                LaunchType.FILE_IMPORT
+                val mimeType = intent.resolveTypeIfNeeded() // Resolve the type if it's a URI
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/IntentHandler.kt:234]: > This is not generally useful for application code; it is used by the frameworks for communicating with back-end system services.

Was this line necessary?

**Review by david-allison — APPROVED** [review]
Clean!

File: AnkiDroid/src/main/java/com/ichi2/anki/IntentHandler.kt (hunk 1 above)
Comment (criticalAY) [reply AnkiDroid/src/main/java/com/ichi2/anki/IntentHandler.kt]: Removed the extra comment

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/IntentHandler.kt]: I was referring to `resolveTypeIfNeeded`, the documentation states the above line: https://developer.android.com/reference/android/content/Intent#resolveTypeIfNeeded(android.content.ContentResolver)

Comment (criticalAY) [reply AnkiDroid/src/main/java/com/ichi2/anki/IntentHandler.kt]: The type causes an issue in older API where the csv extension is misunderstood as file type import hence this, we can rename the function

Comment (criticalAY) [reply AnkiDroid/src/main/java/com/ichi2/anki/IntentHandler.kt]: Intent type was returning the wrong mime type so I created this function

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/IntentHandler.kt]: Oh, I see. It wasn't calling the framework definition. This is fine

**Pull request merged**